python-package/lightgbm/dask.py

```diff
@@ -319,6 +289,9 @@ def _train_part(
         if eval_class_weight:
             kwargs['eval_class_weight'] = [eval_class_weight[i] for i in eval_component_idx]
 
+    if remote_socket is not None:
+        remote_socket.release()
+        time.sleep(0.1)
```

Comment:
I'm not sure how to see the state but I think that's exactly what happened. I saw some failures due to LightGBM not being able to bind the port so I added that to wait a bit, but allowing the reuse seems way better, especially because we're not sending anything through them. I'll make the changes.